### ISSUE TITLE
Consider disallowing setting collision layers and mask without seeing the name

### ISSUE THREAD

**Issue by QbieShay** [godotengine/godot#30834]
**Godot version:**
3.1-stable

**Issue description:**
In the editor it is possible to assign the bitmask of collision layers without seeing the names of the layers (that are visible when pressing on the menu on the right).

While this is good for speed of use, it hides the feature that you can name collision layers and new users might not notice (also users that join a project) 

**Proposal:**
Consider disallowing setting collision layers with the bitmask interface and use only the popup with named layers. I'm not suggesting to disable it completely, but only for collision layers and masks.

**Comment by vortexofdoom** [comment]
This is a related issue, I was wondering about having a collision matrix like Unity's where you specify default interactions between layers. If you know two layers are always/never going to interact it makes sense to be able to set those interactions and then you'd just have to set their layer when you make them.

I think keeping the masks in on a per node basis as overrideable would be good, but it just seems like a good idea to have a "master list" of interactions so you didn't have to reference your other objects when setting collisions. I was going to suggest a similar thing as this post too, as the collision layers are not easily identifiable at a glance.

Basically I think a minor overhaul of the collision/mask layer ui would benefit everyone.

**Comment by madmiraal** [comment]
Feature and improvement proposals for the Godot Engine are now being discussed and reviewed in a dedicated [**Godot Improvement Proposals (GIP)**](https://github.com/godotengine/godot-proposals) ([godotengine/godot-proposals](https://github.com/godotengine/godot-proposals)) issue tracker. The GIP tracker has a detailed issue template designed so that proposals include all the relevant information to start a productive discussion and help the community assess the validity of the proposal for the engine.

The main ([godotengine/godot](https://github.com/godotengine/godot)) tracker is now solely dedicated to bug reports and Pull Requests, enabling contributors to have a better focus on bug fixing work. Therefore, we are now closing all older feature proposals on the main issue tracker.

If you are interested in this feature proposal, please [**open a new proposal on the GIP tracker**](https://github.com/godotengine/godot-proposals/issues/new/choose) following the given issue template (after checking that it doesn't exist already). Be sure to reference this closed issue if it includes any relevant discussion (which you are also encouraged to summarize in the new proposal). Thanks in advance!
